### PR TITLE
[ConnectedMachine] update the module to use API 2022-03-10

### PR DESCRIPTION
**Pull request by dorothyhu**
## Description
Update the API version to stable 2022-03-10
Add cmdlet Update-AzConnectedMachine
Add ResourceGroup to the display table
Fix the issue of extension settings not being able to serialize correctly

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by dorothyhu** [comment]
@VeryEarly Can you take a look at this PR?

File: src/ConnectedMachine/custom/MachineExtensionProperties.json.cs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 using System;
 using Microsoft.Azure.PowerShell.Cmdlets.ConnectedMachine.Runtime.Json;
 
-namespace Microsoft.Azure.PowerShell.Cmdlets.ConnectedMachine.Models.Api20210520
+namespace Microsoft.Azure.PowerShell.Cmdlets.ConnectedMachine.Models.Api20220310
```
Comment (VeryEarly) [inline src/ConnectedMachine/custom/MachineExtensionProperties.json.cs:4]: do you still need those files?

**Comment by VeryEarly** [comment]
format of examples are incorrect, see the previous one.

**Comment by dorothyhu** [comment]
@VeryEarly Fixed the docs issue and added a new cmdlet and the tests. Can you review it again. Thanks

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**